src/routing/pre-match-handler.ts

```diff
@@ -0,0 +1,25 @@
+import type { RequestHandler } from 'express';
+import scheduler from '@milanbot/scheduler';
+import Fixture from '@milanbot/services/fixtures/fixture';
+import PreMatch from '@milanbot//services/thread/pre-match';
+import { FixtureType } from '@milanbot/services/fixtures/types';
+
+const createPreMatchThread = async() => {
+  const prematch = new PreMatch("");
+  await prematch.create();
+};
+
+export const preMatchRequestHandler = (async (req, res, next) => {
+  console.log("Received pre-match request");
+  const handlePreMatchRequest = async () => {
+    const fixtureType = req.body as FixtureType;
+    const fixture = new Fixture(fixtureType);
+    await createPreMatchThread();
+    await scheduler.deletePreMatchCronJob();
+    await scheduler.createMatchCronJob(fixture, `${req.protocol}://${req.get('host')}`);
+    res.status(201);
+    res.send();
+  };
+  handlePreMatchRequest()
+    .catch(next);
+}) as RequestHandler;
```

Comment:
I used promises to avoid wrapping every request handler in a `try catch` block. Any error that occurs will be picked up by the `catch` part of the promise call. Or am I missing something?